tests/bluechi_test/machine.py

```diff
@@ -103,15 +103,24 @@ def systemctl_start_and_wait(self, service_name: str, sleep_after: float = 0.0)
 
         return result, output, wait_result
 
-    def install_systemd_service(self, service: Service,
-                                target_dir: str = os.path.join("/", "etc", "systemd", "system")):
-        LOGGER.debug(f"Installing systemd service '{service.name}' to container path '{target_dir}'"
-                     f"with content:\n{service}")
-        self.create_file(target_dir, service.name, str(service))
+    def install_systemd_service(self, service: Service, backup_existing: bool = False, stop_at_cleanup: bool = True):
+        if backup_existing:
+            self._track_changed_file(service.get_directory(), service.get_name())
+
+        LOGGER.debug(f"Installing systemd service '{service.get_name()}' to container path '{service.get_directory()}'"
+                     f"with content:\n{service.to_string()}")
+        self.create_file(service.get_directory(), service.get_name(), service.to_string())
```

Comment:
I think it would be better to decide if a backup of the file should be created or not in `install_systemd_service`. So I'd remove the `backup_existing` parameter:

```suggestion
        LOGGER.debug(f"Installing systemd service '{service.get_name()}' to container path '{service.get_directory()}'"
                     f"with content:\n{service.to_string()}")
        self.create_file(service.get_directory(), service.get_name(), service.to_string())
```

and enhance the `create_file` function to check if there is already a file for the target path and, if yes, create and track a backup, e.g. like this:
```python
try:
    _, output = self.client.exec_run(f"[ -f {target_file} ] && echo 'exists'")
    if output == "exists":
        self._track_changed_file(target_dir, file_name)
    self.client.create_file(target_dir, file_name, content)
```

Reply:
Done